selectel/config.go

```diff
@@ -32,14 +46,74 @@ func (c *Config) Validate() error {
 			return err
 		}
 	}
+	if c.IdentityEndpoint == "" {
+		c.IdentityEndpoint = DefaultIdentityEndpoint
+	}
 
 	return nil
 }
 
+// Initialize Selectel resell client.
 func (c *Config) resellV2Client() *selvpcclient.ServiceClient {
 	return resellV2.NewV2ResellClientWithEndpoint(c.Token, c.Endpoint)
 }
 
+// Create Keystone token by Selectel token or Keystone credentials.
+func (c *Config) getToken(ctx context.Context, p string, r string) (string, error) {
+	if !c.hasKeystoneCredentials() {
+		return c.getTokenBySelectelToken(ctx, p, r)
+	}
+
+	return c.getTokenByCredentials(ctx, p, r)
+}
+
+// hasKeystoneCredentials determine Keystone credentials exists.
+func (c *Config) hasKeystoneCredentials() bool {
+	return c.User != "" && c.Password != "" && c.DomainName != ""
+}
+
+// Create Keystone token by Selectel token.
+func (c *Config) getTokenBySelectelToken(ctx context.Context, p string, r string) (string, error) {
```

Comment:
Function params can be simplified to 
`(ctx context.Context, p, r string)`

But it seems that we don't need `r string` parameter here, so it can be safely deleted.
Also let's change `p string` to more readable name like `projectID string`

Reply:
Unused region variable removed from all functions - getToken, getTokenBySelectelToken, getTokenByCredentials, getDBaaSClient, getMKSClient, p renamed